presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -6363,4 +6365,32 @@ public void testDependentWindowFunction()
                 "from (select orderkey, orderpriority, totalprice, rank() over(partition by orderpriority order by orderkey) as rnk from orders)";
         assertQuery(sql);
     }
+
+    @Test
+    public void testGroupByLimit()
+    {
+        Session prefilter = Session.builder(getSession())
+                .setSystemProperty(PREFILTER_FOR_GROUPBY_LIMIT, "true")
+                .build();
+        MaterializedResult result1 = computeActual(prefilter, "select count(shipdate), orderkey from lineitem group by orderkey limit 100000");
+        MaterializedResult result2 = computeActual("select count(shipdate), orderkey from lineitem group by orderkey limit 100000");
+        assertEqualsIgnoreOrder(result1, result2, "Prefilter and without prefilter don't give matching results");
+
+        assertQuery(prefilter, "select count(custkey), orderkey from orders where orderstatus='F' and orderkey < 50 group by orderkey limit 100", "values (1, 3), (1, 6), (1, 37), (1, 33), (1, 5)");
+        assertQuery(prefilter, "select count(1) from (select count(custkey), orderkey from orders where orderstatus='F' and orderkey < 50 group by orderkey limit 4)", "select 4");
+        assertQuery(prefilter, "select count(comment), orderstatus from (select upper(comment) comment, upper(orderstatus) orderstatus from orders where orderkey < 50) group by orderstatus limit 100", "values (5, 'F'), (10, 'O')");
+
+        prefilter = Session.builder(getSession())
+                .setSystemProperty(PREFILTER_FOR_GROUPBY_LIMIT, "true")
+                .setSystemProperty(PREFILTER_FOR_GROUPBY_LIMIT_TIMEOUT_MS, "1")
+                .build();
+
+        result1 = computeActual(prefilter, "select count(shipdate), orderkey from lineitem group by orderkey limit 100000");
```

Comment:
can you add tests with GROUP BY + HAVING?
I think these shouldn't be affected by the transformation, just to make sure we don't regress

Reply:
Those tests already exist so I didn't add them. Also the flag is off by default.

Reply:
I meant to make sure we don't cause regressions if the flag is enabled

Reply:
Added one anyway